python-main.js

```diff
@@ -1266,63 +1266,116 @@ function web_editor(config) {
+                window.previousDapWrapper = null;
+            }
         }
 
-        // Disconnect
+        // Disconnect from the microbit
         doDisconnect();
+       
+        var errorType;
+        var errorTitle;
+        var errorDescription;
 
-        // Error handler
-        $("#flashing-overlay-container").css("display", "flex");
-        $("#flashing-info").addClass('hidden');
+        // Determine type of error
+        switch(typeof err) {
+            case "object":
+                console.log("Caught in Promise or Error object");
+                // We might get Error objects as Promise rejection arguments
+                if (!err.message && err.promise && err.reason) {
+                    err = err.reason;
+                }
+                
+                // Determine error type
+                if (err.message === "No valid interfaces found.") {
+                    errorType = "update-req";
+                    errorTitle = err.message;
+                    errorDescription = config["translate"]["webusb"]["err"][errorType];
+                } else if (err.message === "Unable to claim interface.") {
+                    errorType = "clear-connect";
+                    errorTitle = err.message;
+                    errorDescription = config["translate"]["webusb"]["err"][errorType];
+                } else if (err.name === "device-disconnected") {
+                    errorType = "device-disconnected";
+                    errorTitle = err.message;
+                    // No additional message provided here, err.message is enough
+                    errorDescription = "";
+                } else {
+                    // Unhandled error. User will need to reconnect their micro:bit
+                    errorType = "reconnect-microbit";
+                    errorTitle = "WebUSB Error";
+                    errorDescription = config["translate"]["webusb"]["err"][errorType];
+                    if (usePartialFlashing) {
+                        errorDescription += '<br>' + config["translate"]["webusb"]["err"]["partial-flashing-disable"];
+                    } 
+                }
 
-        // We might get Error objects as Promise rejection arguments
-        if (!err.message && err.promise && err.reason) {
-            err = err.reason;
+                break;
+            case "string":
+                // Caught a string. Example case: "Flash error" from DAPjs
+                console.log("Caught a string");
+                
+                // Unhandled error. User will need to reconnect their micro:bit
+                errorType = "reconnect-microbit";
+                errorTitle = "WebUSB Error";
+                errorDescription = config["translate"]["webusb"]["err"][errorType];
+                if (usePartialFlashing) {
+                    errorDescription += '<br>' + config["translate"]["webusb"]["err"]["partial-flashing-disable"];
+                } 
+                break;
```

Comment:
We are not doing anything with the `err` string and just using a generic error title, which is passed to the metrics.
So we will never be able to capture this error in the metrics, and the user sending us screenshots will have generic text that won't give us any indication either.

Can we surface this string in a way that will help us identify issues in this case?

Reply:
Done